src/DotNetty.Common/Concurrency/IEventExecutor.cs

```diff
@@ -46,5 +46,9 @@ public interface IEventExecutor
         Task ShutdownGracefullyAsync();
 
         Task ShutdownGracefullyAsync(TimeSpan quietPeriod, TimeSpan timeout);
+
+        void Schedule(Action<object, object> action, object state1, object state2, TimeSpan delay);
+
+        void Schedule(Action<object, object> action, object state1, object state2, TimeSpan delay, CancellationToken cancellationToken);
```

Comment:
Got it - just read your explanation on #14 

IMHO, add this as a comment:

> To avoid allocations it is often useful to pass more than one object along with work item: this/self and piece of data to be processed. Such an option is available for immediate execution through IEventExecutor.Execute() overload but there's no corresponding version of .Schedule().